nautobot_golden_config/datasources.py

```diff
@@ -125,11 +122,24 @@ def get_id_kwargs(gc_config_item_dict, id_keys, job_result):
         if actual_attr_name in fk_class_mapping:
             if "network_driver" in yaml_attr_name:
                 field_name = "network_driver"
+            elif "platform_name" in yaml_attr_name:
+                field_name = "name"
             else:
                 _, field_name = yaml_attr_name.split("_")
-            kwargs = {field_name: gc_config_item_dict[yaml_attr_name]}
+            if not gc_config_item_dict.get(yaml_attr_name):
+                continue
+            kwargs = {field_name: gc_config_item_dict.get(yaml_attr_name, "")}
             try:
                 id_kwargs[actual_attr_name] = fk_class_mapping[actual_attr_name].objects.get(**kwargs)
+            except fk_class_mapping[actual_attr_name].MultipleObjectsReturned:
+                job_result.log(
```

Comment:
we should add, then use an error code.

Reply:
done, implemented the same pattern from nornir apps. Only defined the two new error codes for now, the rest can be done in another PR.